src/hint_processor/builtin_hint_processor/uint256_utils.rs

```diff
@@ -108,38 +108,53 @@ pub fn uint256_add(
     ap_tracking: &ApTracking,
 ) -> Result<(), HintError> {
     let shift = Felt252::new(1_u32) << 128_u32;
-    let a_relocatable = get_relocatable_from_var_name("a", vm, ids_data, ap_tracking)?;
-    let b_relocatable = get_relocatable_from_var_name("b", vm, ids_data, ap_tracking)?;
-    let a_low = vm.get_integer(a_relocatable)?;
-    let a_high = vm.get_integer((a_relocatable + 1_usize)?)?;
-    let b_low = vm.get_integer(b_relocatable)?;
-    let b_high = vm.get_integer((b_relocatable + 1_usize)?)?;
-    let a_low = a_low.as_ref();
-    let a_high = a_high.as_ref();
-    let b_low = b_low.as_ref();
-    let b_high = b_high.as_ref();
 
-    //Main logic
-    //sum_low = ids.a.low + ids.b.low
-    //ids.carry_low = 1 if sum_low >= ids.SHIFT else 0
-    //sum_high = ids.a.high + ids.b.high + ids.carry_low
-    //ids.carry_high = 1 if sum_high >= ids.SHIFT else 0
+    let a = Uint256::from_var_name("a", vm, ids_data, ap_tracking)?;
+    let b = Uint256::from_var_name("b", vm, ids_data, ap_tracking)?;
+    let a_low = a.low.as_ref();
+    let a_high = a.high.as_ref();
+    let b_low = b.low.as_ref();
+    let b_high = b.high.as_ref();
 
-    let carry_low = if a_low + b_low >= shift {
-        Felt252::one()
-    } else {
-        Felt252::zero()
-    };
+    // Main logic
+    // sum_low = ids.a.low + ids.b.low
+    // ids.carry_low = 1 if sum_low >= ids.SHIFT else 0
+    // sum_high = ids.a.high + ids.b.high + ids.carry_low
+    // ids.carry_high = 1 if sum_high >= ids.SHIFT else 0
+
+    let carry_low = Felt252::from((a_low + b_low >= shift) as u8);
+    let carry_high = Felt252::from((a_high + b_high + &carry_low >= shift) as u8);
 
-    let carry_high = if a_high + b_high + &carry_low >= shift {
-        Felt252::one()
-    } else {
-        Felt252::zero()
-    };
     insert_value_from_var_name("carry_high", carry_high, vm, ids_data, ap_tracking)?;
     insert_value_from_var_name("carry_low", carry_low, vm, ids_data, ap_tracking)
 }
 
+/*
+Implements hint:
+%{
+    res = ids.a + ids.b
+    ids.carry = 1 if res >= ids.SHIFT else 0
+%}
+*/
+pub fn uint128_add(
+    vm: &mut VirtualMachine,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<(), HintError> {
+    let shift = Felt252::new(1_u32) << 128_u32;
+    let a = get_integer_from_var_name("a", vm, ids_data, ap_tracking)?;
+    let b = get_integer_from_var_name("b", vm, ids_data, ap_tracking)?;
+    let a = a.as_ref();
```

Comment:
can you use the as_ref in the variable definition?
`let a = get_integer_from_var_name("a", vm, ids_data, ap_tracking)?.as_ref()`;

Reply:
No. `rustc` complains about dropping the value the reference is referencing 😭